File: 1431-kidsWithTheGreatestNumberOfCandies/kidsWithCandies.hs
kidsWithCandies :: [Int] -> Int -> [Bool]
kidsWithCandies candies extraCandies = map (\x -> x + extraCandies >= maxCandies) candies
    where maxCandies = maximum candies

-- tests

test1 :: Bool
test1 = kidsWithCandies [2,3,5,1,3] 3 == [True,True,True,False,True]

test2 :: Bool
test2 = kidsWithCandies [4,2,1,1,2] 1 == [True,False,False,False,False]

test3 :: Bool
test3 = kidsWithCandies [12,1,12] 10 == [True,False,True]

runTests :: IO ()
runTests = do
    print test1
    print test2
    print test3

File: 1431-kidsWithTheGreatestNumberOfCandies/kids_with_candies.rs
fn kids_with_candies(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
    let max = candies.iter().max().unwrap();
    candies.iter().map(|x| x + extra_candies >= *max).collect()
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test1() {
        assert_eq!(
            kids_with_candies(vec![2, 3, 5, 1, 3], 3),
            vec![true, true, true, false, true]
        );
    }

    #[test]
    fn test2() {
        assert_eq!(
            kids_with_candies(vec![4, 2, 1, 1, 2], 1),
            vec![true, false, false, false, false]
        );
    }

    #[test]
    fn test3() {
        assert_eq!(
            kids_with_candies(vec![12, 1, 12], 10),
            vec![true, false, true]
        );
    }
}
